NtupleProducer/plugins/bRegressionProducer.cc

```diff
@@ -0,0 +1,567 @@
+#include "DataFormats/Common/interface/Handle.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/PatCandidates/interface/Jet.h"
+#include "RecoJets/JetProducers/interface/PileupJetIdAlgo.h"
+#include "DataFormats/PatCandidates/interface/PackedCandidate.h"
+#include "FWCore/Utilities/interface/EDMException.h"
+#include "DataFormats/Math/interface/deltaR.h"
+#include "PhysicsTools/TensorFlow/interface/TensorFlow.h"
+
+#include "RecoVertex/VertexTools/interface/VertexDistance3D.h"
+#include "RecoVertex/VertexPrimitives/interface/ConvertToFromReco.h"
+#include "RecoVertex/VertexPrimitives/interface/VertexState.h"
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+
+#define debug 0
+
+using namespace std;
+using namespace edm;
+
+class bRegressionProducer : public EDProducer
+{
+
+public:
+    bRegressionProducer( const ParameterSet & );
+    ~bRegressionProducer(){};
+    void InitJet();
+    void SetNNVectorVar();
+    std::vector<float> EvaluateNN();
+private:
+    void produce( Event &, const EventSetup & ) override;
+
+    edm::InputTag inputTagJets_;
+    EDGetTokenT<View<pat::Jet> > jetToken_;
+    edm::EDGetTokenT<double> rhoToken_;      
+ 
+    FileInPath bRegressionWeightfile_;
+
+    edm::EDGetTokenT<std::vector<reco::Vertex>> srcVtx_;
+    edm::EDGetTokenT<edm::View<reco::VertexCompositePtrCandidate>> srcSV_;  
+
+    double y_mean_;
+    double y_std_;
+
+    tensorflow::Session* session;
+    std::vector<float> NNvectorVar_; 
+
+
+    //mva variables
+    float Jet_pt ;
+    float Jet_eta ;
+    float rho ;
+    float Jet_mt ;
+    float Jet_leadTrackPt ;
+    float Jet_leptonPtRel ;
+    float Jet_leptonDeltaR ;
+    float Jet_neHEF ;
+    float Jet_neEmEF ;
+    float Jet_vtxPt ;
+    float Jet_vtxMass ;
+    float Jet_vtx3dL ;
+    float Jet_vtxNtrk ;
+    float Jet_vtx3deL ;
+    float Jet_energyRing_dR0_em_Jet_e ;
+    float Jet_energyRing_dR1_em_Jet_e ;
+    float Jet_energyRing_dR2_em_Jet_e ;
+    float Jet_energyRing_dR3_em_Jet_e ;
+    float Jet_energyRing_dR4_em_Jet_e ;
+    float Jet_energyRing_dR0_neut_Jet_e ;
+    float Jet_energyRing_dR1_neut_Jet_e ;
+    float Jet_energyRing_dR2_neut_Jet_e ;
+    float Jet_energyRing_dR3_neut_Jet_e ;
+    float Jet_energyRing_dR4_neut_Jet_e ;
+    float Jet_energyRing_dR0_ch_Jet_e ;
+    float Jet_energyRing_dR1_ch_Jet_e ;
+    float Jet_energyRing_dR2_ch_Jet_e ;
+    float Jet_energyRing_dR3_ch_Jet_e ;
+    float Jet_energyRing_dR4_ch_Jet_e ;
+    float Jet_energyRing_dR0_mu_Jet_e ;
+    float Jet_energyRing_dR1_mu_Jet_e ;
+    float Jet_energyRing_dR2_mu_Jet_e ;
+    float Jet_energyRing_dR3_mu_Jet_e ;
+    float Jet_energyRing_dR4_mu_Jet_e ;
+    float Jet_numDaughters_pt03 ;
+    float Jet_chHEF;//implement from here
+    float Jet_chEmEF;
+    float Jet_leptonPtRelInv;
+    int isEle;
+    int isMu;
+    int isOther;
```

Comment:
Pedantic: keep variables with the same type together.

Reply:
the order of variable is always the same, and iirc has to be respected to fill `NNvectorVar_`